tests/e2e/graphql-test.js

```diff
@@ -0,0 +1,41 @@
+const chai = require('chai');
+const { jestSnapshotPlugin } = require('mocha-chai-jest-snapshot');
+
+chai.use(jestSnapshotPlugin());
+const { expect } = chai;
+
+const prismFactory = require('../helper/prism-loader');
+
+before(() => {
+	global.Prism = prismFactory.createInstance('graphql');
+	require('../../plugins/highlight-keywords/prism-highlight-keywords');
+});
+
+after(() => {
+	delete global['Prism'];
+});
```

Comment:
@RunDevelopment e.g. all of this can be extracted to shared setup so it does not have to be repeated for every new e2e test you might want to add.

The interesting thing /where I was curious for your feedback was how to intelligently `require` plugins while not having to rewrite them... ie. the `before`/`after` hooks.

Reply:
> how to intelligently `require` plugins while not having to rewrite them... ie. the `before`/`after` hooks.

I did it like [this](https://github.com/PrismJS/prism/blob/61739ed731a1a1f4bf3349d5a4de037cedf1deb9/tests/helper/prism-loader.js#L111). I changed the code that load languages to also be able to load plugins. This means that you can do a simple `createInstance(['graphql', 'highlight-keywords'])`.

Reply:
I really like that.